sims/verilator/Makefile

```diff
@@ -116,48 +179,70 @@ model_header_debug = $(model_dir_debug)/V$(VLOG_MODEL).h
+#============================================================================
+# selectively enable verbose logging
+#============================================================================
+# see GNU Make static patterns
+$(sim_vsrcs_printf) : %.printf.v : %.v
+	$(base_dir)/bin/enable_printfs $< $@ "$(ENABLE_PRINTF_PATTERN)"
+
+#============================================================================
 # build makefile fragment that builds the verilator sim rules
-#########################################################################################
-$(model_mk): $(sim_vsrcs) $(sim_common_files) $(EXTRA_SIM_REQS)
+#============================================================================
+$(model_mk): $(sim_vsrcs_printf) $(sim_common_files)
 	rm -rf $(model_dir)
 	mkdir -p $(model_dir)
-	$(VERILATOR) $(VERILATOR_OPTS) -o $(sim) -Mdir $(model_dir) -CFLAGS "-include $(model_header)"
+	$(VERILATOR) $(VERILATOR_OPTS) $(EXTRA_SIM_SOURCES) -o $(sim) \
+		-Mdir $(model_dir) -CFLAGS "-include $(model_header)"
 	touch $@
 
-$(model_mk_debug): $(sim_vsrcs) $(sim_common_files) $(EXTRA_SIM_REQS)
+$(model_mk_debug): $(sim_vsrcs_printf) $(sim_common_files)
 	rm -rf $(model_dir_debug)
 	mkdir -p $(model_dir_debug)
-	$(VERILATOR) $(VERILATOR_OPTS) -o $(sim_debug) --trace -Mdir $(model_dir_debug) -CFLAGS "-include $(model_header_debug)"
+	$(VERILATOR) $(VERILATOR_OPTS) $(EXTRA_SIM_SOURCES) -o $(sim_debug) \
+		$(TRACING_OPTS) \
+		-Mdir $(model_dir_debug) -CFLAGS "-include $(model_header_debug)"
 	touch $@
 
-#########################################################################################
+#============================================================================
 # invoke make to make verilator sim rules
-#########################################################################################
+#============================================================================
 $(sim): $(model_mk) $(dramsim_lib)
 	$(MAKE) VM_PARALLEL_BUILDS=1 -C $(model_dir) -f V$(VLOG_MODEL).mk
 
 $(sim_debug): $(model_mk_debug) $(dramsim_lib)
 	$(MAKE) VM_PARALLEL_BUILDS=1 -C $(model_dir_debug) -f V$(VLOG_MODEL).mk
 
-#########################################################################################
+#============================================================================
 # create a verilator vpd rule
-#########################################################################################
+#============================================================================
 .PRECIOUS: $(output_dir)/%.vpd %.vcd
 $(output_dir)/%.vpd: $(output_dir)/% $(sim_debug)
 	rm -f $@.vcd && mkfifo $@.vcd
 	vcd2vpd $@.vcd $@ > /dev/null &
-	(set -o pipefail && $(sim_debug) $(PERMISSIVE_ON) $(SIM_FLAGS) $(EXTRA_SIM_FLAGS) $(VERBOSE_FLAGS) -v$@.vcd $(PERMISSIVE_OFF) $< </dev/null 2> >(spike-dasm > $<.out) | tee $<.log)
+	(set -o pipefail && $(sim_debug) $(SIM_FLAGS) \
+		$(EXTRA_SIM_FLAGS) $(VERBOSE_FLAGS) \
+		-v$@.vcd $< \
```

Comment:
Please don't inline this flag. It took a long time to DRY this stuff out.

Reply:
inline what flag? i only removed PERMISSIVE

Reply:
You inlined the waveform flag

Reply:
My bad, I thought it was `$(WAVEFORM_FLAG)` before and I think it should be that way.

Reply:
i like to run the debug simulator to see if it passes the first time around, and then i can get the cycle number to start tracing at for the 2nd pass around. i usually get the cycle number from my selectively-enabled printfs.

which would be an argument for NOT including WAVEFORM_FLAG by default

Reply:
You've hardcoded it here, so its always included in this call, I'm suggesting replacing the hardcoded version with `$(WAVEFORM_FLAG)` there are make targets that do not include the waveform flag and those should continue to not include it.

Reply:
oh i didn't see that. so i guess i was arguing to remove the `-v$@.vcd` altogether. 

i'll implement your suggestion

Reply:
actually, i'm just going to leave it. it was like this before, and not part of my PR